lib/api_request.js

```diff
@@ -46,22 +46,18 @@ function sort_by_price_asc (a, b) { // as a user I want to see cheapest holidays
  * @param {Number} hid - the hotel id
  * @param {Function} callback - called once we have a result
  */
-function get_hotel_info (hid, callback) {
-  if (hid && HOTEL_INFO_CACHE[hid]) {
-    return callback(null, HOTEL_INFO_CACHE[hid]);
-  } else {
-    var params = {hotelIds: hid, path: 'hotels'};
-    var options = make_options(params);
-    http_request(options, function (err, data) {
-      if (err) {
-        AwsHelper.log.info({ err: err }, 'Error retrieving hotel info from API');
-        return callback(err, data);
-      } else {
-        HOTEL_INFO_CACHE[hid] = data.result; // cache for next time
-        return callback(err, data.result);
-      }
-    });
-  }
+function get_hotel_info (params, callback) {
+  params.path = 'hotels';
+  var options = make_options(params);
+  AwsHelper.log.trace({ req: options, hotelId: params.hotelId }, 'Fetching hotel info');
+  http.request(options, function (err, data) {
+    /* istanbul ignore if */
+    if (err) {
+      AwsHelper.log.info({ err: err }, 'Error retrieving hotel info from API');
```

Comment:
Does an API error need to be reflected in the UI? 
does calling back with an error here display an error message for the user?